### ISSUE TITLE
Extracting definitions to external files

### ISSUE THREAD

**Issue by cpetrov** [angelozerr/tern-tabris#29]
Opened PR #28 to prepare for the extraction of static definitions to external JSON files. Do you know what would be a good way to include such definition files into the project?

**Comment by angelozerr** [comment]
@cpetrov I'm sorry I don't understand?

**Comment by cpetrov** [comment]
The idea is to separate definitions data from the plugin logic. We thought externalizing the data by splitting the `defs["!define"]` object into smaller objects, putting them into separate files and including them in tabris.js would be a good idea. E.g. CommonJS style:

```javascript
var defs = {
  "!name": "tabris",
  "!define": {
    "!propertyTypes": {
      "Bounds": require("./propertyTypes/Bounds.json"),
      "Color": require("./propertyTypes/Color.json"),
      ...
    },
    "!properties": {
      "ButtonProperties": require("./properties/Button.json"),
      "PageProperties": require("./properties/Button.json")
    },
    "!events": {...},
    "types": {...}
  },
  "tabris": {...}
}
```

For example Browserify can be used to build a JS bundle out of the modules. Do you think this is a good idea?

**Comment by angelozerr** [comment]
In my case I develop plugin with Chrome (with browser) without webserver (file system, so AJAX cannot be used to load some JSON file). I'm afraid that it will be boring for the development.

**Comment by cpetrov** [comment]
A tool like browserify can be used to create a bundle JavaScript file out of all modules during build. Then loading of separate modules should not be a concern.

**Comment by angelozerr** [comment]
> A tool like browserify can be used to create a bundle JavaScript file out of all modules during build. 

Yes but I'm afraid for development mode, it will be boring (generates each time with browserfy as soon as you change your code). But if you think it's very very important, do it.
